crates/pathfinder/src/observability.rs

```diff
@@ -0,0 +1,56 @@
+use futures::Future;
+use prometheus::{Encoder, TextEncoder};
+use std::net::SocketAddr;
+use std::pin::Pin;
+use warp::http::Response as HttpResponse;
+use warp::hyper::StatusCode;
+use warp::Filter;
+
+type ServerFuture = Pin<Box<dyn Future<Output = ()>>>;
+
+/// Run metrics server if `addr` is specified.
+pub fn run_server(addr: Option<SocketAddr>) -> anyhow::Result<(ServerFuture, Option<SocketAddr>)> {
+    if let Some(addr) = addr {
+        let exporter = opentelemetry_prometheus::exporter().init();
+        let readyz = warp::path!("readyz").and(warp::get()).map({
+            move || {
+                HttpResponse::builder()
+                    .status(StatusCode::OK)
+                    .body("ok")
+                    .unwrap()
+            }
+        });
+
+        let livez = warp::path!("livez").and(warp::get()).map({
+            move || {
+                HttpResponse::builder()
+                    .status(StatusCode::OK)
+                    .body("ok")
+                    .unwrap()
+            }
+        });
+        let metrics = warp::path!("metrics").and(warp::get()).map({
+            move || {
+                let mut buffer = Vec::new();
+                let encoder = TextEncoder::new();
```

Comment:
I will update it to use `opentelemetry_prometheus` only, I didn't notice the re-exports.

I'm using opentelemetry so that in the future metrics can be exported to other protocols, e.g. to an opentelemetry endpoint together with traces.